File: code_blocks/integrations/third-party-integrations/mixpanel_1.swift
import SwiftUI
import RevenueCat
import Mixpanel

@main
struct MyApp: App {
    init() {
        Purchases.configure(withAPIKey: "public_sdk_key", appUserID: "my_app_user_id")
        Mixpanel.mainInstance().identify(distinctId: "my_app_user_id")
        Purchases.shared.attribution.setMixpanelDistinctID(Mixpanel.mainInstance().distinctId)
    }

    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

File: code_blocks/integrations/third-party-integrations/mixpanel_1.kt
import android.app.Application
import com.mixpanel.android.mpmetrics.MixpanelAPI
import com.revenuecat.purchases.Purchases

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        Purchases.configure(this, "public_sdk_key", appUserID = "my_app_user_id")
        val mixpanel = MixpanelAPI.getInstance(this, "MIXPANEL_TOKEN")
        mixpanel.identify("my_app_user_id")
        Purchases.sharedInstance.attribution.setMixpanelDistinctId(mixpanel.distinctId)
    }
}
